pkg/virt-handler/vm_test.go

```diff
@@ -309,6 +309,28 @@ var _ = Describe("VirtualMachineInstance", func() {
 			diskutils.MockDefaultOwnershipManager()
 		})
 
+		It("should do nothing if vmi and domain do not exist", func() {
+			controller.Execute()
+			testutils.ExpectEvent(recorder, "")
```

Comment:
just so we're clear; without your fix, this test would fail, right ?

Reply:
Yes, In my environment, after the handler is restarted, it calls execute, but the cached does not get the vmi.

Reply:
@stu-gott these are unit tests - located in `pkg/...` ; e2e tests would be in `test/...` 

Am I missing something ?

Reply:
yes, correct.

Reply:
Sorry, I haven't had time to deal with this these days, I'll take another look when I get some time.